packages/testcontainers/src/containers/RegTestContainer/index.ts

```diff
@@ -37,11 +37,27 @@ export class RegTestContainer extends DeFiDContainer {
     ]
   }
 
+  async waitForBlockHeight (height: number, timeout = 590000): Promise<void> {
+    throw new Error('RegTestContainer Violation: waitForBlockHeight is not implemented for non master node containers')
+  }
+
+  async waitForWalletBalanceGTE (balance: number, timeout = 300000): Promise<void> {
+    throw new Error('RegTestContainer Violation: waitForWalletBalanceGTE is not implemented for non master node containers')
+  }
+
   async getNewAddress (label: string = '', addressType: 'legacy' | 'p2sh-segwit' | 'bech32' | string = 'bech32'): Promise<string> {
     return await this.call('getnewaddress', [label, addressType])
   }
 
   async getRpcPort (): Promise<string> {
     return await this.getPort('19554/tcp')
   }
+
+  async generate (nblocks: number, address?: string, maxTries: number = 1000000): Promise<void> {
+    throw new Error('RegTestContainer Violation: generate is not implemented for non master node containers')
+  }
+
+  async fundAddress (address: string, amount: number): Promise<{ txid: string, vout: number }> {
+    throw new Error('RegTestContainer Violation: fundAddress is not implemented for non master node containers')
+  }
```

Comment:
Commenting on this file as a whole, I don't think we should be needed to introduce this sort of methods. And we don't strictly need these for testing in our packages outside of jellyfish-testing since the returned type of Container is correctly type on instantiation.

```
const testing = Testing.create(new RegTestContainer())
```

Will correctly only display functionality available to it.

That however doesn't work internally in jellyfish-testing if we look at the https://github.com/DeFiCh/jellyfish/blob/kodemon/testing-regtest-support/packages/jellyfish-testing/src/testing.ts#L17-L22 part of testing instance since these sub dependencies assumes the container is `MasterNodeRegTestContainer`.

So we may need to adjust the sub dependencies to be flexible and not make container type assumptions.